### PR TITLE
Upgrade to pytorch 1.6

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
(Almost) get us to pytorch 1.6. Turns out we covered most of it already. Additionally sort of needs #3025.

**Testing steps**
CI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -89,47 +89,51 @@ installdeps: &installdeps
...
-      python -m pip install --progress-bar off 'torch==1.3.1'
-      python -m pip install --progress-bar off 'subword-nmt==0.3.7'
-      python -m pip install --progress-bar off pytorch-pretrained-bert
-      python -m pip install --progress-bar off transformers
-      python -m pip install --progress-bar off fairseq
+      python -m pip install --progress-bar off 'torch==1.5.1'
+      python -c 'import torch; print("Torch version:", torch.__version__)'
+      python -m torch.utils.collect_env
+
+installtorchgpu14: &installtorchgpu14
```
Comment (EricMichaelSmith) [inline .circleci/config.yml:113]: Hmm we still want to support 1.4 instead of just supporting the last 2 versions?

File: parlai/utils/testing.py [hunk 2]
```diff
@@ -30,6 +30,13 @@
     TORCH_AVAILABLE = False
     GPU_AVAILABLE = False
 
+try:
+    import torchvision  # noqa: F401
+
+    VISION_AVAILABLE = True
+except ImportError:
+    VISION_AVAILABLE = False
```
Comment (EricMichaelSmith) [inline parlai/utils/testing.py:38]: Hmm should we log this or no?

File: requirements.txt [hunk 3]
```diff
@@ -8,7 +8,7 @@ flake8==3.7.8
 fvcore==0.1.1.post20200716
 gitdb2==2.0.5
 GitPython==3.0.3
-torch==1.4.0
+torch
```
Comment (EricMichaelSmith) [inline requirements.txt:11]: no `torch>=1.4.0`?

File: parlai/utils/testing.py (hunk 2 above)
Comment (stephenroller) [reply parlai/utils/testing.py:38]: Nah,  the test says "skipped because"

File: .circleci/config.yml (hunk 1 above)
Comment (stephenroller) [reply .circleci/config.yml:113]: I don't want to force the entire team to upgrade right now.

**Pull request merged**